packages/common/src/resources/upsertResource.ts

```diff
@@ -0,0 +1,89 @@
+import { IUserRequestOptions } from "@esri/arcgis-rest-auth";
+import {
+  addItemResource,
+  getItemResources,
+  updateItemResource,
+} from "@esri/arcgis-rest-portal";
+import HubError from "../HubError";
+import { objectToJsonBlob, stringToBlob } from ".";
+import { getPortalApiUrl } from "../urls";
+
+/**
+ * Given an item, and owner, Search for if the resource exists
+ * and if does, update it, otherwise add it. Returns a url for the item.
+ *
+ * @export
+ * @param {string} id
+ * @param {string} owner
+ * @param {*} resource
+ * @param {string} name
+ * @param {IUserRequestOptions} ro
+ * @param {string} [prefix=""]
+ * @return {*}  {Promise<string>}
+ */
+export async function upsertResource(
+  id: string,
+  owner: string,
+  resource: any,
+  name: string,
+  ro: IUserRequestOptions,
+  prefix: string = ""
+): Promise<string> {
+  try {
+    const extension = name.split(".").pop();
+    // Search against the item resources to see if the resource exists
+    const doesResourceExist: boolean = await getItemResources(id, ro).then(
```

Comment:
I could be reading this wrong, but I don't actually think this is returning a boolean right now. The `.find` will return the first element in the resources array that has the resource name (if any)...so I think it's either returning a resource object or undefined